systemdev/testing-locally.rst

```diff
@@ -16,6 +16,179 @@ We only recommend developing packages using a device with Ubuntu Touch installed
+
+    chroot_base=/srv/chroot/ubports-focal-amd64
+    sudo sbuild-createchroot --components=main,restricted,universe --extra-repository='deb http://archive.ubuntu.com/ubuntu/ focal-updates main restricted universe' --include=ccache focal "${chroot_base}" http://archive.ubuntu.com/ubuntu/
+
+The UBports package repository needs to be added using::
+
+    wget 'http://repo.ubports.com/keyring.gpg' -O - | sudo tee "${chroot_base}/usr/share/keyrings/ubports-keyring.gpg" >/dev/null
+    printf 'deb [signed-by=/usr/share/keyrings/ubports-keyring.gpg] http://repo2.ubports.com/ focal main\n' | sudo tee "${chroot_base}/etc/apt/sources.list.d/ubports.list" >/dev/null
+    sbuild-update -u -d focal
+
+Maintaining build chroots
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Synchronizing package index files and subsequent package upgrades can be performed using::
+
+    sbuild-update -u -d focal
+
+Optimizations
+^^^^^^^^^^^^^
+
+Caching package downloads
+"""""""""""""""""""""""""
+
+In order to save bandwidth and time it is highly advisable to cache downloaded packages by using apt-cacher-ng.  It can be installed with::
+
+    apt install apt-cacher-ng
+
+Chroots then need to be configured so that apt inside the chroot uses apt-cacher-ng on the host as a proxy server::
+
+    printf 'Acquire::http { Proxy "http://localhost:3142"; }\n' | sudo tee "${chroot_base}/etc/apt/apt.conf.d/proxy" >/dev/null
+
+Caching compilation results
+"""""""""""""""""""""""""""
+
+ccache is a compiler cache which speeds up repeated compilation of the same source code by caching the resulting object files. The actual cache is stored on the host system and bind-mounted into sbuild chroots with a schroot hook::
+
+    ccache_dir=/var/cache/ccache-sbuild
+    sudo install --group=sbuild --mode=2775 -d "${ccache_dir}"
+    sudo env CCACHE_DIR="${ccache_dir}" ccache --max-size 4G
+    printf '%s %s none rw,bind 0 0\n' "${ccache_dir}" "${ccache_dir}" | sudo tee -a /etc/schroot/sbuild/fstab >/dev/null
+
+In order to make use of ccache inside a sbuild chroot a wrapper script needs to be created::
+
+    cat >"${ccache_dir}/sbuild-ccache.sh" <<EOF
+    #!/bin/sh
+    export CCACHE_DIR=$ccache_dir
+    export CCACHE_UMASK=002
+    export CCACHE_COMPRESS=1
+    unset CCACHE_HARDLINK
+    export PATH=/usr/lib/ccache:\$PATH
+    exec "\$@"
+    EOF
+    chmod +x "${ccache_dir}/sbuild-ccache.sh"
+
+In order to use this wrapper script the following line must be added to the configuration of a schroot chroot in `/etc/schroot/chroot.d/`::
+
+    command-prefix=/var/cache/ccache-sbuild/sbuild-ccache.sh
+
+Building a package
+^^^^^^^^^^^^^^^^^^
+
+sbuild needs to be invoked from a debianized package source directory, UBports packages mostly consists of git repositories containing native or non-native packages.  The following script can be used in order to prepare a non-native package for a build::
```

Comment:
what is "debianized" ? what does native/non-native mean in this context? would it make sense to give this part a different heading "preparing blaa"/"debianizing blub"/"native somethingoranother..." and move the section heading "Building a package" down to only the actual build step.